### PR TITLE
Fix fill in the course selector

### LINKED ISSUE #235: When navigating to /projects/create?course_id=x fill in the course selector

### LINKED ISSUE #260: use loader in project form

### PR DESCRIPTION
**Pull request by warreprovoost**
Added a loader to fetch the projects
Closes #235 and closes #260

### CODE REVIEW

File: frontend/src/components/ProjectForm/ProjectForm.tsx [hunk 1]
```diff
@@ -83,9 +78,22 @@ export default function ProjectForm() {
...
-  useEffect(() => {
-    fetchCourses();
-  }, [regexError]);
+  const courses = useLoaderData() as Course[]
+
+  const location = useLocation();
+  const urlParams = new URLSearchParams(location.search);
+  const initialCourseId = urlParams.get('course_id') || '';
+  let initialCourseName = ''
+  for( const c of courses){
```
Comment (AronBuzogany) [inline frontend/src/components/ProjectForm/ProjectForm.tsx:87]: Wrap this in an effect, or this will be run on every key/mouse pressed on this page

**Review by AronBuzogany — CHANGES_REQUESTED** [review]

File: frontend/src/utils/fetches/fetch-courses.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { authenticatedFetch } from "../authenticated-fetch.ts";
+const API_URL = import.meta.env.VITE_APP_API_HOST;
+
+export const fetchCourses = async (user_uid: string) => {
```
Comment (AronBuzogany) [inline frontend/src/utils/fetches/fetch-courses.ts:4]: define it as a function

File: frontend/src/utils/fetches/fetch-courses.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    const response = await authenticatedFetch(
+      `${API_URL}/courses?teacher=${user_uid}`,
```
Comment (AronBuzogany) [inline frontend/src/utils/fetches/fetch-courses.ts:7]: `teacher=` not needed, should be handled by authentication

File: frontend/src/utils/fetches/fetch-courses.ts
Comment (AronBuzogany) [inline frontend/src/utils/fetches/fetch-courses.ts:1]: this doesn't need its own file, make a general file for `courses`

File: frontend/src/utils/fetches/fetch-project-form.ts
Comment (AronBuzogany) [inline frontend/src/utils/fetches/fetch-project-form.ts:1]: same for this, doesn't need its own file

**Review by AronBuzogany — CHANGES_REQUESTED** [review]

File: frontend/src/utils/fetches/fetch-courses.ts (hunk 3 above)
Comment (warreprovoost) [reply frontend/src/utils/fetches/fetch-courses.ts:7]: if `teacher=` is gone then all the courses of that user are visible not just the one where it is a teacher of

File: frontend/src/utils/fetches/fetch-courses.ts
Comment (warreprovoost) [reply frontend/src/utils/fetches/fetch-courses.ts:1]: I renamed the file and also the function so it is more clear what does, the `loader` component asks to not put it in a file with a lot of other components hence why I gave it it's own

File: frontend/src/utils/fetches/fetch-courses.ts (hunk 3 above)
Comment (AronBuzogany) [reply frontend/src/utils/fetches/fetch-courses.ts]: Correct, my BAD

**Review by AronBuzogany — APPROVED** [review]

**Pull request merged**